test/coretasks/test_coretasks_sasl.py

```diff
@@ -179,6 +293,33 @@ def test_sasl_plain_not_supported(botfactory: BotFactory, tmpconfig) -> None:
     ), 'SASL mech is not available so we must stop here.'
 
 
+def test_sasl_plain_nonempty_server_message(
+    botfactory: BotFactory,
+    tmpconfig,
+) -> None:
+    mockbot = botfactory.preloaded(tmpconfig, preloads=['coretasks'])
+    mockbot.backend.connected = True
+
+    # connect and capability negotiation
+    mockbot.on_connect()
+    mockbot.on_message(':irc.example.com CAP * LS :sasl=PLAIN,EXTERNAL')
+    n = len(mockbot.backend.message_sent)
+    mockbot.on_message(':irc.example.com CAP * ACK :sasl')
+
+    # sanity check
+    assert mockbot.backend.message_sent[n:] == rawlist(
+        'AUTHENTICATE PLAIN',
+    )
+
+    # server acknowledges PLAIN auth request in an unexpected way
+    n = len(mockbot.backend.message_sent)
+    mockbot.on_message(':irc.example.com AUTHENTICATE VGVzdEJvdABUZXN0Qm90AG')
+    assert mockbot.backend.message_sent[n:] == rawlist(
+        'AUTHENTICATE *',
+    ), ('Bot must abort SASL PLAIN auth if server reply to starting '
+        'AUTHENTICATE PLAIN is not as expected per SASL spec')
```

Comment:
What is the expected behavior from the server here? Because I'd make sure here we can send CAP END after that cancelation.

Reply:
> The client can abort an authentication by sending an asterisk as the data. The server will send a 906 numeric.

906 (`ERR_SASLABORTED`) is handled by `sasl_fail()`. Sopel will log the failure, then quit with a "SASL Auth Failed" message.

Reply:
Have done so!